featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -1195,3 +1208,84 @@ def test_primitive_options_multiple_inputs(es):
+                                    primitive_options=options2)
+    features2 = dfs_obj2.build_features()
+
+    assert set(features2) == set(features1)
+
+
+def test_primitive_options_class_names(es):
+    options1 = {
+        'mean': {'include_entities': ['customers']}
+    }
+
+    options2 = {
+        Mean: {'include_entities': ['customers']}
+    }
+
+    dfs_obj1 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=['mean'],
+                                    trans_primitives=[],
+                                    primitive_options=options1)
+    features1 = dfs_obj1.build_features()
+
+    dfs_obj2 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=['mean'],
+                                    trans_primitives=[],
+                                    primitive_options=options2)
+    features2 = dfs_obj2.build_features()
+
+    dfs_obj3 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=[Mean],
+                                    trans_primitives=[],
+                                    primitive_options=options1)
+    features3 = dfs_obj3.build_features()
+
+    dfs_obj4 = DeepFeatureSynthesis(target_entity_id='cohorts',
+                                    entityset=es,
+                                    agg_primitives=[Mean],
+                                    trans_primitives=[],
+                                    primitive_options=options2)
+    features4 = dfs_obj4.build_features()
+
+    for f in features1:
+        deps = f.get_dependencies(deep=True)
+        entities = [d.entity.id for d in deps]
+        if isinstance(f.primitive, Mean):
+            assert all(entity == 'customers' for entity in entities)
+
+    assert set(features1) == set(features2) == set(features3) == set(features4)
+
+
+def test_primitive_options_instantiated_primitive(es):
+    skipna_mean = Mean(skipna=False)
+    options = {
+        skipna_mean: {'include_entities': ['stores']},
+        'mean': {'ignore_entities': ['stores']}
+    }
+    dfs_obj = DeepFeatureSynthesis(target_entity_id='régions',
+                                   entityset=es,
+                                   agg_primitives=['mean', skipna_mean],
+                                   trans_primitives=[],
+                                   primitive_options=options)
+
+    features = dfs_obj.build_features()
+    for f in features:
+        deps = f.get_dependencies(deep=True)
+        entities = [d.entity.id for d in deps]
+        if f.primitive == skipna_mean:
+            assert all(entity == 'stores' for entity in entities)
+        elif isinstance(f.primitive, Mean):
+            assert 'stores' not in entities
```

Comment:
Should we also add a test to make sure an error is raised if conflicting options are specified using a combination of primitive type and string name? Seems to be working correctly, but I don't think we explicitly verify that in our tests.